File: src/main/java/com/dozycoffee/wms/warehouse/application/port/in/result/WarehouseResult.java
package com.dozycoffee.wms.warehouse.application.port.in.result;

import com.dozycoffee.wms.warehouse.domain.enumeration.AvailabilityStatus;
import com.dozycoffee.wms.warehouse.domain.model.Warehouse;

import java.math.BigDecimal;

public record WarehouseResult(
        Long warehouseId,
        String warehouseName,
        String address,
        BigDecimal latitude,
        BigDecimal longitude,
        AvailabilityStatus warehouseStatus
) {

    public static WarehouseResult from(Warehouse warehouse) {
        return new WarehouseResult(
                warehouse.getWarehouseId(),
                warehouse.getWarehouseName(),
                warehouse.getAddress().value(),
                warehouse.getCoordinate().latitude(),
                warehouse.getCoordinate().longitude(),
                warehouse.getWarehouseStatus()
        );
    }
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/application/port/in/result/WarehouseResult.kt
package com.dozycoffee.wms.warehouse.application.port.`in`.result

import com.dozycoffee.wms.warehouse.domain.enumeration.AvailabilityStatus
import com.dozycoffee.wms.warehouse.domain.model.Warehouse
import java.math.BigDecimal

data class WarehouseResult(
    val warehouseId: Long,
    val warehouseName: String,
    val address: String,
    val latitude: BigDecimal,
    val longitude: BigDecimal,
    val warehouseStatus: AvailabilityStatus
) {
    companion object {
        fun from(warehouse: Warehouse): WarehouseResult {
            return WarehouseResult(
                requireNotNull(warehouse.warehouseId),
                warehouse.warehouseName,
                warehouse.address.value,
                warehouse.coordinate.latitude,
                warehouse.coordinate.longitude,
                warehouse.warehouseStatus
            )
        }
    }
}
